src/wallet/wallet.cpp

```diff
@@ -1310,7 +1278,7 @@ isminetype CWallet::IsMine(const CTxOut &txout) const {
             return ISMINE_NO;
         }
 
-        return db.HasZerocoinEntry(pub) ? ISMINE_SPENDABLE : ISMINE_NO;
+        return CZerocoinState::GetZerocoinState()->HasCoin(pub) ? ISMINE_SPENDABLE : ISMINE_NO;
```

Comment:
If the check is performed before ConnectBlock the answer is always "no". If the check is performed after ConnectBlock the answer is always "yes". Doesn't seem right to me

Reply:
How this solves wallet swelling?